### PR TITLE
improvement(health_checker): add a retry to health validator

### PR DESCRIPTION
**Pull request by enaydanov**
Trello: https://trello.com/c/tSVvqcWH/2122-add-a-retry-to-the-health-validator

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2393,21 +2394,46 @@ def run_nodetool(self, sub_cmd, args="", options="", timeout=None,
...
         # Disable it
         if not self.parent_cluster.params.get('cluster_health_check'):
             return
 
-        nodes_status = self.get_nodes_status()
-        peers_details = self.get_peers_info() or {}
-        gossip_info = self.get_gossip_info() or {}
+        # Make several attempts to validate node health.
+        # All attempts except last will break after the first issue to retry again.
+        for publish in (False, ) * (retries - 1) + (True, ):
```
Comment (bentsi) [inline sdcm/cluster.py:2405]: too complex, better to simlify to:
```python
error_events = []
for _ in range(retries):
    ...
    error_events = list(itertools.chain(...))
    if error_events:
        time.sleep(CHECK_NODE_HEALTH_RETRY_DELAY)
        continue
    else:
           break

    
if error_events:
    for event in events:
        event.publish()


```

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2393,21 +2394,46 @@ def run_nodetool(self, sub_cmd, args="", options="", timeout=None,
...
+            nodes_status = self.get_nodes_status()
+            peers_details = self.get_peers_info() or {}
+            gossip_info = self.get_gossip_info() or {}
+
+            events = itertools.chain(
+                check_nodes_status(
```
Comment (bentsi) [inline sdcm/cluster.py:2411]: call with kw arguments

File: sdcm/cluster.py (hunk 1 above)
Comment (enaydanov) [reply sdcm/cluster.py:2405]: This will break the idea to stop on the first validation error.

Comment (enaydanov) [reply sdcm/cluster.py]: simplified

File: sdcm/cluster.py (hunk 2 above)
Comment (enaydanov) [reply sdcm/cluster.py]: fixed

File: sdcm/cluster.py [hunk 3]
```diff
@@ -2393,21 +2394,39 @@ def run_nodetool(self, sub_cmd, args="", options="", timeout=None,
...
-                           removed_nodes_list=self.parent_cluster.dead_nodes_ip_address_list)
-        check_node_status_in_gossip_and_nodetool_status(gossip_info, nodes_status, current_node=self)
-        check_schema_version(gossip_info, peers_details, nodes_status, current_node=self)
-        check_nulls_in_peers(gossip_info, peers_details, current_node=self)
+            nodes_status = self.get_nodes_status()
+            peers_details = self.get_peers_info() or {}
+            gossip_info = self.get_gossip_info() or {}
+
+            events = itertools.chain(
+                check_nodes_status(nodes_status, self, self.parent_cluster.dead_nodes_ip_address_list),
```
Comment (bentsi) [inline sdcm/cluster.py:2411]: convert to KW arguments

**Review by bentsi — APPROVED** [review]

**Pull request merged**